### PR TITLE
Appset deployment using pull model

### PR DESCRIPTION
**Pull request by am-agrawa**
Starting ACM 2.10, appset gitops provides an option to choose between pull and push model, where push model was default so far. Starting ODF 4.16, we would recommend and switch to using pull model with ACM 2.10 and above versions.

### CODE REVIEW

File: tests/functional/disaster-recovery/regional-dr/test_failover_and_relocate.py [hunk 1]
```diff
@@ -95,6 +117,8 @@ def test_failover_and_relocate(
         pass the yaml conf/ocsci/dr_ui.yaml to trigger it.
         The value of lastGroupSyncTime is verified in each stage.
 
+        appset_model pull type was introduced with ACM 2.10 and would be supported with ODF 4.16 and above
```
[the remark this answers is not in the document]
Comment (am-agrawa) [reply tests/functional/disaster-recovery/regional-dr/test_failover_and_relocate.py:126]: Done

File: ocs_ci/deployment/deployment.py [hunk 2]
```diff
@@ -40,6 +40,7 @@
     get_lvm_full_version,
     check_cephcluster_status,
 )
+from ocs_ci.ocs.constants import CLUSTERROLEBINDING_APPSET_PULLMODEL_PATH
```
[the remark this answers is not in the document]
Comment (am-agrawa) [reply ocs_ci/deployment/deployment.py]: done

File: tests/conftest.py [hunk 3]
```diff
@@ -6483,6 +6484,8 @@ def factory(
             pvc_interface (str): 'CephBlockPool' or 'CephFileSystem'.
                 This decides whether a RBD based or CephFS based resource is created. RBD is default.
             switch_ctx (int): The cluster index by the cluster name
+            appset_model (str): Appset Gitops deployment now supports "pull" model starting ACM 2.10 in addition
+                to default "push" model. Default value is set to None.
```
[the remark this answers is not in the document]
Comment (am-agrawa) [reply tests/conftest.py]: changed to default pull

File: ocs_ci/deployment/deployment.py [hunk 4]
```diff
@@ -284,9 +284,13 @@ def do_gitops_deploy(self):
 
         # Multicluster operations
         if config.multicluster:
-            config.switch_acm_ctx()
+            # Gitops operator is needed on all clusters for appset type workload deployment using pull model
+            for cluster_index in range(config.multicluster):
```
[the remark this answers is not in the document]
Comment (am-agrawa) [reply ocs_ci/deployment/deployment.py]: done

File: ocs_ci/deployment/deployment.py [hunk 5]
```diff
@@ -334,6 +337,13 @@ def do_gitops_deploy(self):
             gitops_obj._has_phase = True
             gitops_obj.wait_for_phase("successful", timeout=720)
 
+            logger.info(
+                "Create clusterrolebinding on both the managed clusters needed "
+                "for appset pull model gitops deployment"
+            )
+            for _ in managed_clusters:
+                run_cmd(f"oc create -f {CLUSTERROLEBINDING_APPSET_PULLMODEL_PATH}")
```
[the remark this answers is not in the document]
Comment (am-agrawa) [reply ocs_ci/deployment/deployment.py]: done

**Comment by am-agrawa** [comment]
All test passed on ODF 4.16- https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-rdr-setup/733/

**Review by prsurve — APPROVED** [review]

**Review by keesturam — APPROVED** [review]

**Pull request merged**